Worker does not find 3rd party deps with Bazel 6 --experimental_allow_unresolved_symlinks

In https://github.com/aspect-build/bazel-examples:

```shell
$ export USE_BAZEL_VERSION=last_green    
$ cd next.js
$ bazel run start --experimental_allow_unresolved_symlinks

INFO: Analyzed target //:start (0 packages loaded, 1550 targets configured).
INFO: Found 1 target...
ERROR: /Users/matt/Documents/GitHub/bazel-examples/next.js/pages/api/BUILD.bazel:5:11: Compiling TypeScript project @//pages/api:api_typings [tsc -p tsconfig.json] failed: (Exit 1): tsc_worker.sh failed: error executing command (from target //pages/api:api_typings) bazel-out/darwin-opt-exec-2B5CBBC6/bin/external/npm_typescript/tsc_worker.sh @bazel-out/darwin-fastbuild/bin/pages/api/hello.d.ts-0.params

pages/api/hello.ts(2,54): error TS2307: Cannot find module 'next' or its corresponding type declarations.

--- a/.bazelrc.common
+++ b/.bazelrc.common
@@ -77,7 +77,3 @@ build:openbsd --nolegacy_external_runfiles
 # on the host.
 # See https://github.com/angular/angular/issues/27514.
 build --incompatible_strict_action_env
-
-# Not yet supported with rules_ts
-# See https://github.com/aspect-build/rules_ts/pull/202
-build --noexperimental_allow_unresolved_symlinks

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -73,18 +73,11 @@ jobs:
                     # Don't test bzlmod with Bazel 5 (not supported)
                     - bazelversion: 5.3.2
                       folder: e2e/bzlmod
-                    # TODO: RBE requires --experimental_allow_unresolved_symlinks which is blocked on
-                    # https://github.com/aspect-build/rules_ts/pull/202
-                    - config: rbe
-                    # TODO: fix worker_workspace with Bazel 6
-                    - bazelversion: 6.0.0rc1
-                      folder: e2e/worker_workspace
                     # TODO: e2e/worker needs a bit of to be configured for RBE
                     - config: rbe
                       folder: e2e/worker
                     - config: rbe
                       folder: e2e/worker_workspace
-
         # Steps represent a sequence of tasks that will be executed as part of the job
         steps:
             # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
@@ -146,4 +139,19 @@ jobs:
               working-directory: ${{ matrix.folder }}
               # hashFiles returns an empty string if test.sh is absent
               if: ${{ hashFiles(format('{0}/test.sh', matrix.folder)) != '' }}
-              run: ./test.sh
+              run: |
+                  cp -f $GITHUB_WORKSPACE/.github/workflows/ci.bazelrc .bazelrc.test
+                  echo "build --config=${{ matrix.config }}" >> .bazelrc.test
+                  echo "build --disk_cache=" >> .bazelrc.test
+                  ./test.sh
+
+            - name: run ./test.sh with --noexperimental_allow_unresolved_symlinks
+              working-directory: ${{ matrix.folder }}
+              # hashFiles returns an empty string if test.sh is absent
+              if: ${{ hashFiles(format('{0}/test.sh', matrix.folder)) != '' }}
+              run: |
+                  cp -f $GITHUB_WORKSPACE/.github/workflows/ci.bazelrc .bazelrc.test
+                  echo "build --config=${{ matrix.config }}" >> .bazelrc.test
+                  echo "build --disk_cache=" >> .bazelrc.test
+                  echo "build --noexperimental_allow_unresolved_symlinks" >> ".bazelrc.test"
+                  ./test.sh

--- a/.gitignore
+++ b/.gitignore
@@ -1,3 +1,4 @@
 bazel-*
 .bazelrc.user
+.bazelrc.test
 node_modules/

--- a/e2e/worker/.bazelrc
+++ b/e2e/worker/.bazelrc
@@ -1,4 +1,7 @@
 # import common bazelrc shared with e2e workspaces
 import %workspace%/../../.bazelrc.common
 
+# used by test.sh to invoke bazel with different configurations
+try-import %workspace%/.bazelrc.test
+
 build:verbose --define=VERBOSE_LOGS=1

--- a/e2e/worker/test.sh
+++ b/e2e/worker/test.sh
@@ -146,7 +146,7 @@ for dep in "${deps[@]}"; do
     buildozer "remove deps //:node_modules/$dep" :ts
     message="error TS2307: Cannot find module '$dep' or its corresponding type declarations."
     bazel build :ts 2>&1 | grep "$message" || exit_with_message "Case 11: expected worker to report \"$message\""
-    buildozer "add deps //:node_modules/$dep"
+    buildozer "add deps //:node_modules/$dep" :ts
 done
 
 

--- a/e2e/worker_workspace/.bazelrc
+++ b/e2e/worker_workspace/.bazelrc
@@ -1,0 +1,5 @@
+# import common bazelrc shared with e2e workspaces
+import %workspace%/../../.bazelrc.common
+
+# used by test.sh to invoke bazel with different configurations
+try-import %workspace%/.bazelrc.test

--- a/ts/private/ts_project_worker.js
+++ b/ts/private/ts_project_worker.js
@@ -101,10 +101,22 @@ function createFilesystemTree(root, inputs) {
             }
             notifyWatchers(dirs, parts.base, TYPE.SYMLINK, EVENT_TYPE.ADDED);
         } else if (parts.base) {
-            node[parts.base] = {
-                [Type]: TYPE.FILE
+            const new_node = {
+                [Type]: TYPE.FILE,
             }
-            notifyWatchers(dirs, parts.base, TYPE.FILE, EVENT_TYPE.ADDED);
+            try {
+                const linkPath = path.join(root, p)
+                const readlink = fs.readlinkSync(linkPath);
+                const targetPath = path.isAbsolute(readlink) ? readlink : path.join(path.dirname(linkPath), readlink)
+                const relative = path.relative(root, targetPath);
+                if (relative != p) {
+                    new_node[Type] = TYPE.SYMLINK
+                    new_node[Symlink] = relative
+                }
+            } catch (e) { /* Can't determine if it's a symlink. move on */ }
+
+            node[parts.base] = new_node;
+            notifyWatchers(dirs, parts.base, new_node[Type], EVENT_TYPE.ADDED);
         }
     }
 

